src/Features/LanguageServer/Protocol/Extensions/Extensions.cs

```diff
@@ -24,6 +23,13 @@ public static Uri GetURI(this TextDocument document)
             return ProtocolConversions.GetUriFromFilePath(document.FilePath);
         }
 
+        public static Document? GetDocument(this ILspSolutionProvider provider, TextDocumentIdentifier textDocument, string? clientName)
+        {
+            var (documentId, solution) = provider.FindDocumentAndSolution(textDocument, clientName);
```

Comment:
reads weird that this is called FindDocument, but returns a DocumentId.  Why not actually just return a document?